terraform/api/k8s/main.tf

```diff
@@ -101,7 +101,7 @@ resource "kubernetes_deployment" "api" {
         share_process_namespace         = true
 
         dynamic "image_pull_secrets" {
-          for_each = var.docker_hub_authentication != null ? [var.docker_hub_authentication] : []
+          for_each = var.docker_hub_authentication != "NULL" ? [var.docker_hub_authentication] : []
```

Comment:
Thanks Nahid. Were you able to find an explanation on why was this causing the error?